nncf/quantization/algorithms/weight_compression/torch_fx_backend.py

```diff
@@ -0,0 +1,273 @@
+            if weight is None or not isinstance(weight, Tensor):
+                raise nncf.InternalError(f"Could not find a nncf.tensor in the model by name {weight_name}.")
+
+            # calculates compressed weights and decompression parameters
+            compressed_weight = compress_weight(
+                weight,
+                wc_params.reduction_axes,
+                compression_config,
+                None if precomputed_scales is None else precomputed_scales.get(wc_params.weight_name),
+                None if precomputed_zero_points is None else precomputed_zero_points.get(wc_params.weight_name),
+            )
+            compressed_weight.scale = compressed_weight.scale.astype(dtype=TensorDataType.float16)
+
+            # pack compressed tensor
+            if compression_config.mode == CompressWeightsMode.INT8_SYM:
+                dtype = TensorDataType.int8
+            else:
+                dtype = TensorDataType.uint8
+            packed_tensor = compressed_weight.tensor.astype(dtype)
+
+            self.set_weight(wc_params.node_with_weight, wc_params.weight_port_id, model, graph, packed_tensor)
+
+            if len(consumer_nodes) > 1:
+                raise nncf.InternalError("Shared weights not supported in compression for TorchFX models")
+
+            # creates weight decompressor
+            if compression_config.mode == CompressWeightsMode.INT8_SYM:
+                decompressor = SymmetricWeightsDecompressor(
+                    compressed_weight.scale.data, result_dtype=weight.data.dtype
+                )
+                decompressor_type = "symmetric"
+            else:
+                packed_zero_point = compressed_weight.zero_point.astype(dtype)
+                decompressor = AsymmetricWeightsDecompressor(
+                    compressed_weight.scale.data, packed_zero_point.data, result_dtype=weight.data.dtype
+                )
+                decompressor_type = "asymmetric"
+
+            # registry weight decompression module in the model
+            # TODO: Find a more efficient way to access updated constant name
+            compressed_weight_name = wc_params.node_with_weight.node_name + "_updated_constant0"
+            decompressor_name = f"{decompressor_type}_weights_decompressor_{compressed_weight_name.replace('.', '_')}"
+
+            # inserts the weight decompressor into the model as the post hook on the model weight
+            transformation_layout.register(
+                FXApplyTransformationCommand(
+                    module_insertion_transformation_builder(
+                        decompressor,
+                        [PTTargetPoint(TargetType.OPERATOR_POST_HOOK, target_node_name=compressed_weight_name)],
+                        decompressor_name,
+                    )
+                )
+            )
+
+        # apply transformations
+        transformed_model = FXModelTransformer(model).transform(transformation_layout)
```

Comment:
Is it possible to apply transformation as soon as a layer is compressed?
Seems like we will be switching to another design of transform function. @alexsu52 
https://github.com/openvinotoolkit/nncf/pull/2816#discussion_r1689764923

Reply:
This is possible.

Right now, it does not give any benefits because weights already compressed and transformations only insert decompression operations.